sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3281,6 +3282,22 @@ def teardown_no_lock(self,
         config = common_utils.read_yaml(handle.cluster_yaml)
         cluster_name = handle.cluster_name
         use_tpu_vm = config['provider'].get('_has_tpus', False)
+
+        # Use the new provisioner for AWS.
+        if isinstance(cloud, clouds.AWS):
+            # TODO(suquark): An alternative approach is to move cloud
+            #  termination logic into cloud classes (for example,
+            #  "clouds.AWS" here). Currently I just use the code from
+            #  the new provisioner, which I believe benefits further
+            #  development of "SkyPilot as a service". Let's see how
+            #  this goes.
+            region = config['provider']['region']
+            provision_lib.teardown_cluster(repr(cloud), region, cluster_name,
+                                           terminate)
+            if post_teardown_cleanup:
+                self.post_teardown_cleanup(handle, terminate, purge)
+            return
```

Comment:
This is great!
More I am thinking about this, I am wondering if we can have all those operations that reach the cloud providers using their APIs to be located within the `provision` module (may need to rename the `provision` to another name, if we decide to go that way). That said, I am thinking to move the `cloud.query_status` to `provision` module as well, so that to leave the `Cloud` as a frontend class for operating/querying the catalogs, and `provision` module to be the backend for each cloud.